lsp/nls/src/linearization/mod.rs

```diff
@@ -24,7 +24,7 @@ pub mod building;
 pub mod completed;
 pub mod interface;
 
-pub type Environment = nickel_lang::environment::Environment<Ident, usize>;
+pub type Environment = nickel_lang::environment::Environment<(Ident, FileId), usize>;
```

Comment:
I wonder why you need `FileId` to be an input of your table. My hunch would be that it should be an output, that is, at any point of the program, we have an environment that maps the identifiers of the current scope to a linearization id. This id may be in other files (typically e.g. for the stdlib). Which would give:

```rust
pub type Environment = nickel_lang::environment::Environment<Ident, (FileId, usize)>;
```

In fact, I would even set `pub type ItemID = (FileId, usize)` and use that everywhere instead of `usize`. So that gotodefinition, etc. can cross files.

As it is now, you need to provide a `FileId` before getting information from an identifier. As you can't guess where this identifier comes from, what can you pass beside the current `FileId`?

Reply:
I think I misunderstood your initial suggestion